File: app/src/main/java/com/blackmirror/dongda/model/DetailInfoServiceImagesBean.java
package com.blackmirror.dongda.model;

import android.support.annotation.NonNull;

import com.blackmirror.dongda.utils.StringUtils;

/**
 * Created by xcx on 2018/5/20.
 */

public class DetailInfoServiceImagesBean implements Comparable<DetailInfoServiceImagesBean>{
    /**
     * tag : 8
     * image : 597ec6a9-79c6-4a6f-806f-f5a58f9629f5
     */

    public String tag;
    public String image;

    @Override
    public int compareTo(@NonNull DetailInfoServiceImagesBean o) {
        if (!StringUtils.isNumber(this.tag) || !StringUtils.isNumber(o.tag)){
            return -1;
        }
        if (Integer.parseInt(this.tag)>=Integer.parseInt(o.tag)){
            return 1;
        }else {
            return -1;
        }
    }
}

File: app/src/main/java/com/blackmirror/dongda/model/DetailInfoServiceImagesBean.kt
package com.blackmirror.dongda.model

import com.blackmirror.dongda.utils.isNumber

/**
 * Created by xcx on 2018/5/20.
 */

class DetailInfoServiceImagesBean : Comparable<DetailInfoServiceImagesBean> {
    /**
     * tag : 8
     * image : 597ec6a9-79c6-4a6f-806f-f5a58f9629f5
     */

    var tag: String? = null
    var image: String? = null

    override fun compareTo(o: DetailInfoServiceImagesBean): Int {
        if (tag == null || o.tag == null || !tag!!.isNumber() || !o.tag!!.isNumber()) {
            -1
        }
        return if (tag!!.toInt() >= o.tag!!.toInt()) {
            1
        } else {
            -1
        }
    }
}
